### PR TITLE
HPCC-23982 Suppress forceRemoteRead (dafilesrv access) in containers

### PR DESCRIPTION
**Pull request by jakesmith**
Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by jakesmith** [comment]
At some point it may be better to suppress the logic in each engine that can potentially perform remote reads in a cloud setup, I think this is a pragmatic change for now, and only really affects the regression tests.

@richardkchapman - please review

File: common/workunit/workunit.cpp [hunk 1]
```diff
@@ -6314,6 +6314,13 @@ void CLocalWorkUnit::loadPTree(IPropertyTree *ptree)
 {
     clearCached(false);
     p.setown(ptree);
+#ifdef _CONTAINERIZED
+    if (getDebugValueBool("forceRemoteRead", false))
+    {
+        WARNLOG("Remote reading via dafilesrv not supposed within containers - disabling");
```
Comment (richardkchapman) [inline common/workunit/workunit.cpp:6320]: typo - supposed should say supported.

Not sure it's really worth warning though?

**Comment by richardkchapman** [comment]
@jakesmith One minor comment.

File: common/workunit/workunit.cpp (hunk 1 above)
Comment (jakesmith) [reply common/workunit/workunit.cpp:6320]: ok, yes wasn't sure. I'll remove the warning.

**Comment by jakesmith** [comment]
@richardkchapman - corrected.

**Comment by richardkchapman** [comment]
I'm a teeny bit concerned this will lead to confusion at some point - it's an effective quick fix for now but...

@ghalliday What do you think?

**Comment by jakesmith** [comment]
@ghalliday - moved suppression into engines.
Please review.

**Pull request merged**